validation/src/main/scala/hmda/validation/rules/lar/validity/V290.scala

```diff
@@ -1,36 +1,19 @@
 package hmda.validation.rules.lar.validity
 
-import hmda.model.census.CBSATractLookup
 import hmda.model.fi.lar.LoanApplicationRegister
 import hmda.validation.dsl.Result
 import hmda.validation.rules.EditCheck
 import hmda.validation.dsl.PredicateCommon._
 import hmda.validation.dsl.PredicateSyntax._
+import hmda.validation.dsl.PredicateGeo._
 
 object V290 extends EditCheck[LoanApplicationRegister] {
 
-  val cbsaTracts = CBSATractLookup.values
-
-  val validCombinations = cbsaTracts.map { cbsa =>
-    (cbsa.state, cbsa.county, cbsa.geoIdMsa)
-  }.toSet
-
-  val validMdCombinations = cbsaTracts.map { cbsa =>
-    (cbsa.state, cbsa.county, cbsa.metDivFp)
-  }.toSet
-
   override def name: String = "V290"
 
-  override def apply(input: LoanApplicationRegister): Result = {
-    val msa = input.geography.msa
-    val state = input.geography.state
-    val county = input.geography.county
-
-    val combination = (state, county, msa)
-
-    when(msa not equalTo("NA")) {
-      (combination is containedIn(validCombinations)) or
-        (combination is containedIn(validMdCombinations))
+  override def apply(lar: LoanApplicationRegister): Result = {
+    when(lar.geography.msa not equalTo("NA")) {
+      lar.geography is validStateCountyMsaCombination
```

Comment:
I'm somewhat less convinced of this part (just this one line, I mean -- the rest is good). I think "valid" is a bit vague in that predicate, because it's specifically about this one rule, V290.  I don't know whether I'd prefer just a different naming, or two predicates, or what. But I do feel like the logic is a bit obscured. (You may be able to convince me it's fine, though.)